### DISCUSSION TITLE
Can it run MySQL/MariaDB

### DISCUSSION THREAD

**Discussion by vabene1111** [TandoorRecipes/recipes#246] [Q&A] [answered]
Can this application be run with a MySQL/MariaDB Database ?

**Comment by vabene1111** [comment, marked as answer]
> **Due to time restraints running this application with anything other than postgresql is not officially supported (meaning i don't test it and i probably cant help you with your problems). This Tutorial is written without any proper testing but should generally lead the way. If anyone is interested in creating a proper tutorial/testing this feel free to get back to me so we can improve it.**

> If you are using MySQL for compatibility reasons Sqlite might be the better choice (as long as your setup is relativly small and you make sure to properly backup the sqlite file)

Django nattily supports MariaDB/MySQL as a Storage backend. The recommended option is to use `mysqlclient` which needs to be installed.

Please read the [offical docs](https://docs.djangoproject.com/en/3.1/ref/databases/#mariadb-notes) for more information on how to setup and configure mysql.

In order to run with mysql you will need to change the `.env` file to point to the database server (and select the appropriate engine)
```
DB_ENGINE=django.db.backends.mysql
POSTGRES_HOST=<mysql-server-ip>
POSTGRES_PORT=<mysql-server-port>
POSTGRES_USER=<mysql-user>
POSTGRES_PASSWORD=<mysql-user-password>
POSTGRES_DB=<name of the database>
```

You will also need to manually build your container with the `mysqlclient` library. For this clone the repository, add `mysqlclient` to the `requirments.txt` and run `docker build` on the directoy.

**Reply by Ronaldvr** [reply]
OK, I tested it, and got it to work. The build file needs:

     RUN apk add --no-cache postgresql-libs gettext zlib libjpeg libxml2-dev libxslt-dev mysql-client mariadb-connector-c-dev mariadb-dev 

Additionally And indeed as you said 
    mysqlclient==2.0.3
 in the requirements.txt 

The env file adaptation works as expected.

These are indeed all that is required.

I would be grateful if you could add these to the standard build

**Reply by vabene1111** [reply]
I have thought a lot about this in the last few weeks. I always try to give every possible option so everyone has the biggest possible freedom. 

But each option i give makes it harder and harder for me to develop this application further and to give support for every way of installing this. 

I am not sure yet what to do but for now there wont be any new officially supported installation methods.

**Reply by TheUltimateC0der** [reply]
What's the basic problem by supporting multiple RDBMS ?
Are you using any kind of PostgreSQL specific features?

Not looked into the code yet, but by using a ORM like Django, supporting multiple RDBMS should be pretty easy.

**Reply by smilerz** [reply]
The search is built using SearchVectors that are Postgress specific.  Between that and needing to setup, maintain and test against multiple database types is time consuming.

**Reply by ddubnyk** [reply]
> > **Due to time restraints running this application with anything other than postgresql is not officially supported (meaning i don't test it and i probably cant help you with your problems). This Tutorial is written without any proper testing but should generally lead the way. If anyone is interested in creating a proper tutorial/testing this feel free to get back to me so we can improve it.**
> 
> > If you are using MySQL for compatibility reasons Sqlite might be the better choice (as long as your setup is relativly small and you make sure to properly backup the sqlite file)
> 
> Django nattily supports MariaDB/MySQL as a Storage backend. The recommended option is to use `mysqlclient` which needs to be installed.
> 
> Please read the [offical docs](https://docs.djangoproject.com/en/3.1/ref/databases/#mariadb-notes) for more information on how to setup and configure mysql.
> 
> In order to run with mysql you will need to change the `.env` file to point to the database server (and select the appropriate engine)
> 
> ```
> DB_ENGINE=django.db.backends.mysql
> POSTGRES_HOST=<mysql-server-ip>
> POSTGRES_PORT=<mysql-server-port>
> POSTGRES_USER=<mysql-user>
> POSTGRES_PASSWORD=<mysql-user-password>
> POSTGRES_DB=<name of the database>
> ```
> 
> You will also need to manually build your container with the `mysqlclient` library. For this clone the repository, add `mysqlclient` to the `requirments.txt` and run `docker build` on the directoy.

I'm sorry if i slightly misunderstand you but i'm attempting to get this working and i'm getting stuck.  Its running in podman, but it worked fine before i started to try move to an external Database.  So I don't think this has any impact on the situation.  I feel it must be something simple i'm doing wrong that i'm simply missing.

Command I run to start the container:

`podman run --log-level debug --name Recipes --network Recipes -p 8063:8080 -v /var/podman/recipes/config:/opt/recipes/staticfiles -v /var/podman/recipes/recipefiles:/opt/recipes/mediafiles --env "/var/podman/recipes/.env" localhost/recipes`

I've built the container with the msqlclient package and have it starting, however it keeps saying:

`[WARNING]
The environment variable 'SECRET_KEY' is not set but REQUIRED for running Tandoor!
Waiting for database to be ready...
[WARNING]
The environment variable 'POSTGRES_PASSWORD' is not set but REQUIRED for running Tandoor!`

It then shuts down.
However, I have indeed set this in the environment file.

Here is the section in the file.  I understand you don't officially support it, but i'm more curious if i'm simply missing some configuration or something simple.

``# random secret key, use for example `base64 /dev/urandom | head -c50` to generate one
# ---------------------------- REQUIRED -------------------------
SECRET_KEY=>^KvMP^v2yxxxxxxxxxxxxx
# ---------------------------------------------------------------

# your default timezone See https://timezonedb.com/time-zones for a list of timezones
TIMEZONE=America/Vancouver

# add only a database password if you want to run with the default postgres, otherwise change settings accordingly
DB_ENGINE=django.db.backends.mysql
# DB_OPTIONS= {} # e.g. {"sslmode":"require"} to enable ssl
POSTGRES_HOST=192.168.53.6
POSTGRES_PORT=3306
POSTGRES_USER=recipes
# ---------------------------- REQUIRED -------------------------
POSTGRES_PASSWORD=">$<*QU='=tRxxxxxxxxxxx"
# ---------------------------------------------------------------
POSTGRES_DB=recipes``
